hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/utils/TracingContext.java

```diff
@@ -153,6 +153,24 @@ public static String validateClientCorrelationID(String clientCorrelationID) {
     return clientCorrelationID;
   }
 
+  public void setPrimaryRequestIDBlob() {
+    // Generate a UUID, remove dashes, and shuffle by selecting a random 8-character substring
+    String uuid = UUID.randomUUID().toString().replace("-", "");
+    StringBuilder randomId = new StringBuilder(8);
+
+    for (int i = 0; i < 8; i++) {
+      int randomIndex = (int) (Math.random() * uuid.length());
+      randomId.append(uuid.charAt(randomIndex));
+    }
+
+    primaryRequestId = randomId + "B";
```

Comment:
I think you need to convert randomId to a String before assigning it to primaryRequestId. Something like this:
primaryRequestId = randomId.toString() + "B";

Reply:
This is not needed as we are already adding a string